File: CSE3063F22_iteration2_classes/Log.java
import java.io.FileReader;
import java.io.FileWriter;
import java.io.IOException;

import org.json.simple.JSONObject;
import org.json.simple.parser.JSONParser;
import org.json.simple.parser.ParseException;

public class Log {
private String key;

public Log(){
  
}

public String getKey(){
    return key;
}
    public static void create_logFile(String semester) throws IOException{  
   /*The reason it is static is that we get an 
   error when we define the log in the 
   courseregistrationsimulation class. */
    JSONObject jsonObject = new JSONObject();
       int def = 0;
       jsonObject.put("SemesterError", def);
       jsonObject.put("GPAError", def);
       jsonObject.put("PrerequisiteError", def);
       jsonObject.put("SeatLimitError", def);
       FileWriter file = new FileWriter("errorLog_"+semester+".json");
       file.write(jsonObject.toJSONString());
       file.close();
}

    public void logging_error(String key, String semester) throws IOException, ParseException{
       
            Object obj = new JSONParser().parse(new FileReader("errorLog_"+semester+".json"));
            JSONObject jo = (JSONObject) obj;
            String value = String.valueOf(jo.get(getKey()));
           
            int new_value = Integer.parseInt(value) + 1;
           
            jo.remove(key);
            jo.put(key, new_value);
          
            FileWriter writer = new FileWriter(("errorLog_"+semester+".json"), false); //overwrites the content of file
            writer.write(jo.toString());
            writer.close();
         
            
        
    }
    
}

File: ITERATION3/OOSD_3RD_ITERATİON_code/Log.py
import json

class Log:
    

    def create_logFile(semester):
        json_object = {
            "SemesterError": 0,
            "GPAError": 0,
            "PrereqError": 0,
            "SeatLimitError": 0,
            "AlreadyPassedError": 0
        }
        with open(f"errorLog_{semester}.json", "w") as file:
            json.dump(json_object, file)

    def logging_error(self, key, semester):
        with open(f"errorLog_{semester}.json", "r") as file:
            json_object = json.load(file)
        value = json_object[key]
        new_value = value + 1
        json_object[key] = new_value
        
        with open(f"errorLog_{semester}.json", "w") as file:
            json.dump(json_object, file)

    def write_txt(self, semester):
        with open(f"errorLog_{semester}.json", "r") as file:
            json_string = json.dumps(json.load(file))
        
        with open(f"errorLog_{semester}.txt", "w") as file:
            file.write(json_string)
